core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/CarbonUIUtil.java

```diff
@@ -502,4 +537,70 @@ public static String removeTenantSpecificStringsFromURL(String requestURL) {
     private static Object getDefaultHomePageProductParam() {
         return getProductParam(CarbonConstants.PRODUCT_XML_WSO2CARBON + CarbonConstants.DEFAULT_HOME_PAGE);
     }
+
+    /**
+     * Returns absolute URL of admin console webapp for given relative path
+     * if IS_RESOLVE_ABSOLUTE_URLS_ENABLED config is enabled.
+     *
+     * @param context       Webapp context root of the Carbon webapp.
+     * @param relativePath  Relative path of the Carbon webapp
+     * @param request       Request that used to redirect.
+     * @return absolute URL of admin console webapp for given relative path.
+     */
+    public static String resolveAdminConsoleBaseURL(String context, String relativePath, HttpServletRequest request) {
+
+        if (isResolveAbsoluteURLsEnabled()) {
+
+            // Removing any tailing "/" in the context.
+            context = getAdminConsoleBaseURL(context);
+            if (context.endsWith("/")) {
+                context = context.substring(0, context.length() - 1);
+            }
+
+            // Remove any tailing "/carbon" in context to build base URL.
+            if (context.endsWith("/carbon")) {
```

Comment:
Can’t we use a constant for `/carbon` and then reference it here? Also, when getting the length in line 562, we can use var.length(), which improves maintainability.